test/test_repolib.py

```diff
@@ -852,6 +862,147 @@ def test_releasever_the_string_none(self):
         # you laugh, but someone thinks that would be an awesome product name.
         self.assertTrue(YumReleaseverSource.is_set({'releaseVer': 'None'}))
 
+class AptRepoFileTest(unittest.TestCase):
+
+    def test_includes_yum_parameters(self):
+        # Setup mock
+        ar = AptRepoFile()
+        yr = YumRepoFile()
+        mock_repo = Repo('mock', [('baseurl', 'https://example.org.org/'),
+                                  ('sslclientcert', 'mypem.pem')])
+        # Modify data using different backends (Apt,Yum)
+        ar_fixed_content = ar.fix_content(mock_repo)
+        yr_fixed_content = yr.fix_content(mock_repo)
+        # Yum parameters must be included in Apt as well (garuanty compatibility)
+        for key in yr_fixed_content:
+            self.assertIn(key, ar_fixed_content)
+
+    def test_globals(self):
+        # Setup expected results
+        exp_path = 'etc/apt/sources.list.d'
+        exp_name = 'rhsm.sources'
+        exp_content_types = ['deb']
+        exp_repofile_header = """#
+# Certificate-Based Repositories
+# Managed by (rhsm) subscription-manager
+#
+# *** This file is auto-generated.  Changes made here will be over-written. ***
+# *** Use "subscription-manager repo-override --help" if you wish to make changes. ***
+#
+# If this file is empty and this system is subscribed consider
+# a "apt-get update" to refresh available repos
+#
+# *** DO NOT EDIT THIS FILE ***
+#
+"""
+        ar = AptRepoFile()
+        # Test actual data against expected results
+        self.assertEqual(ar.PATH, exp_path)
+        self.assertEqual(ar.NAME, exp_name)
+        self.assertEqual(ar.CONTENT_TYPES, exp_content_types)
+        self.assertEqual(ar.REPOFILE_HEADER, exp_repofile_header)
+
+    def test_add_repo(self):
+        # Setup mock
+        ar = AptRepoFile()
+        mock_repo = Repo('mock')
+        mock_dict = dict([(str(k), str(v)) for (k, v) in mock_repo.items()])
+        mock_deb822 = StubDeb822(mock_dict)
+        # Modify data
+        ar.add(mock_repo)
+        # Test modification
+        act_repo = ar.repos822[0]
+        for key in mock_deb822:
+            self.assertIn(key, act_repo)
+            self.assertEqual(act_repo[key], mock_repo[key])
+
+    def test_delete_repo(self):
+        # Setup mock
+        ar = AptRepoFile()
+        mock_repo_a = Repo('a')
+        mock_repo_b = Repo('b')
+        mock_repo_c = Repo('c')
+        ar.add(mock_repo_a)
+        ar.add(mock_repo_b)
+        ar.add(mock_repo_c)
+        # Modify data
+        ar.delete(mock_repo_b)
+        # Test modification
+        self.assertEqual(len(ar.repos822), 2)
+        for repo822 in ar.repos822:
+            self.assertTrue('a' == repo822['id'] or 'c' == repo822['id'])
```

Comment:
IMHO a slightly cleaner solution can be to sort the repositories by id, so it is easier to check each by equality

Reply:
Implemented as suggested.